### PR TITLE
Smoke CO: Improve profile selection

### PR DESCRIPTION
**Pull request by nickolas-deboom**
The Heiman CO sensor is a new matter 1.2 device that utilizes the smoke/CO alarm subdriver. It currently matches to a profile that contains capabilities unsupported by the device - smokeDetector, relativeHumidityMeasurement, and temperatureMeasurement. This PR contains a new profile for this device as well as restructures the naming scheme of the profiles for this subdriver to be more scalable so that it will be easier to match future smoke/CO detection devices to profiles.

Note that the existing profiles will be left unchanged for now, but will be removed when drivers for matter 1.2 are released. New profiles containing the same capabilities as these existing profiles were also created, with the new naming scheme.

### CODE REVIEW

**Review by ctowns — APPROVED** [review]

File: drivers/SmartThings/matter-sensor/src/smoke-co-alarm/init.lua [hunk 1]
```diff
@@ -34,26 +35,79 @@ local function is_matter_smoke_co_alarm(opts, driver, device)
...
+  "smoke-co-temp-humidity-COmeas"
+}
+
 local function match_profile(device)
   local smoke_eps = device:get_endpoints(clusters.SmokeCoAlarm.ID, {feature_bitmap = clusters.SmokeCoAlarm.types.Feature.SMOKE_ALARM})
   local co_eps = device:get_endpoints(clusters.SmokeCoAlarm.ID, {feature_bitmap = clusters.SmokeCoAlarm.types.Feature.CO_ALARM})
+  local temp_eps = device:get_endpoints(clusters.TemperatureMeasurement.ID)
+  local humidity_eps = device:get_endpoints(clusters.RelativeHumidityMeasurement.ID)
+  local CO_meas_eps = device:get_endpoints(clusters.CarbonMonoxideConcentrationMeasurement.ID, {feature_bitmap = clusters.CarbonMonoxideConcentrationMeasurement.types.Feature.NUMERIC_MEASUREMENT})
+  local CO_level_eps = device:get_endpoints(clusters.CarbonMonoxideConcentrationMeasurement.ID, {feature_bitmap = clusters.CarbonMonoxideConcentrationMeasurement.types.Feature.LEVEL_INDICATION})
```
Comment (samadDotDev) [inline drivers/SmartThings/matter-sensor/src/smoke-co-alarm/init.lua:62]: nit: even though CO is an acronym, we should remain consistent in the lower snake_case (as previously done for `co_eps`) for variables here

```suggestion
  local co_eps = device:get_endpoints(clusters.SmokeCoAlarm.ID, {feature_bitmap = clusters.SmokeCoAlarm.types.Feature.CO_ALARM})
  local temp_eps = device:get_endpoints(clusters.TemperatureMeasurement.ID)
  local humidity_eps = device:get_endpoints(clusters.RelativeHumidityMeasurement.ID)
  local co_meas_eps = device:get_endpoints(clusters.CarbonMonoxideConcentrationMeasurement.ID, {feature_bitmap = clusters.CarbonMonoxideConcentrationMeasurement.types.Feature.NUMERIC_MEASUREMENT})
  local co_level_eps = device:get_endpoints(clusters.CarbonMonoxideConcentrationMeasurement.ID, {feature_bitmap = clusters.CarbonMonoxideConcentrationMeasurement.types.Feature.LEVEL_INDICATION})
```

File: drivers/SmartThings/matter-sensor/profiles/co-COmeas.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: co-COmeas
```
Comment (samadDotDev) [inline drivers/SmartThings/matter-sensor/profiles/co-COmeas.yml:1]:
```suggestion
name: co-comeas
```

or maybe

```suggestion
name: co-coMeas
```

**Review by samadDotDev — APPROVED** [review]

File: drivers/SmartThings/matter-sensor/src/smoke-co-alarm/init.lua (hunk 1 above)
Comment (nickolas-deboom) [reply drivers/SmartThings/matter-sensor/src/smoke-co-alarm/init.lua]: Updated as suggested in f80ab53ddfdaf9987592ed6702cc1d2a543fb364.

File: drivers/SmartThings/matter-sensor/profiles/co-COmeas.yml (hunk 2 above)
Comment (nickolas-deboom) [reply drivers/SmartThings/matter-sensor/profiles/co-COmeas.yml]: Updated as suggested in f80ab53ddfdaf9987592ed6702cc1d2a543fb364.

**Pull request merged**